### PR TITLE
Add rolling restart page

### PR DESCRIPTION
**Pull request by kbatuigas**
This PR creates a new standalone guide for performing a rolling restart for a RP cluster. This content is taken from the existing [Upgrade RP Linux - Perform a rolling upgrade](https://docs.redpanda.com/current/upgrade/rolling-upgrade/#perform-a-rolling-upgrade) page. 

This allows us to cross reference rolling restart content in pages such as the [new whole cluster recovery doc](https://github.com/redpanda-data/docs/pull/118), where the upgrade step isn't necessary or relevant.

Since the steps to do a rolling restart and rolling upgrade overlap, the content is single sourced with this PR. New attributes have been added to the Upgrade RP Linux and the new Rolling Restart pages (`:rolling-upgrade:` and `:rolling-restart:` respectively) so the partials can conditionally check whether the page describes a restart vs upgrade, and display the appropriate text. 

**Preview:** [Rolling restart](https://deploy-preview-143--redpanda-docs-preview.netlify.app/current/manage/cluster-maintenance/rolling-restart/)
**Preview:** [Upgrade Redpanda - Perform a rolling upgrade](https://deploy-preview-143--redpanda-docs-preview.netlify.app/current/upgrade/rolling-upgrade/#perform-a-rolling-upgrade)

Closes https://github.com/redpanda-data/documentation-private/issues/2128

### CODE REVIEW

File: modules/manage/pages/cluster-maintenance/rolling-restart.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+= Rolling Restart
```
Comment (andrwng) [inline modules/manage/pages/cluster-maintenance/rolling-restart.adoc:1]: We should also include https://github.com/redpanda-data/docs/blob/main/modules/upgrade/partials/rolling-upgrades/restart-impact.adoc

File: modules/manage/pages/cluster-maintenance/rolling-restart.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+include::upgrade:partial$rolling-upgrades/check-metrics.adoc[]
+
+include::upgrade:partial$rolling-upgrades/restart-the-broker.adoc[]
+
+include::upgrade:partial$rolling-upgrades/disable-maintenance-mode.adoc[]
+
+include::upgrade:partial$rolling-upgrades/post-upgrade-tasks.adoc[]
+
+include::shared:partial$suggested-reading.adoc[]
```
Comment (andrwng) [inline modules/manage/pages/cluster-maintenance/rolling-restart.adoc]: All this reusable content is neat!

**Review by andrwng — COMMENTED** [review]
Overall LGTM!

**Review by andrwng — APPROVED** [review]

**Review by vuldin — APPROVED** [review]
Thanks! LGTM

**Comment by micheleRP** [comment]
@JakeSCahill would you please check this PR? This new/separate [Rolling Restart page](https://deploy-preview-143--redpanda-docs-preview.netlify.app/current/manage/cluster-maintenance/rolling-restart/) in the nav tree is for Linux. Will this be confusing for anyone doing a [K8s rolling upgrade](https://deploy-preview-143--redpanda-docs-preview.netlify.app/current/upgrade/k-rolling-upgrade/#perform-a-rolling-upgrade)? If you'll want tabs on this new page for Linux and Kubernetes, will that be a separate PR?

File: modules/upgrade/partials/rolling-upgrades/enable-maintenance-mode.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+```bash
+rpk cluster health --watch --exit-when-healthy
+```
++
+The combination of the `--watch` and `--exit-when-healthy` flags tell rpk to monitor the cluster health and exit only when the cluster is back in a healthy state.
++
+[NOTE]
+====
+ifdef::rolling-upgrade[]
+You can also evaluate xref:manage:monitoring.adoc[external metrics] to determine cluster health. If the cluster has any issues, take the broker out of maintenance mode by running the following command before proceeding with other operations, such as decommissioning or retrying the rolling upgrade:
```
Comment (micheleRP) [inline modules/upgrade/partials/rolling-upgrades/enable-maintenance-mode.adoc]:
```suggestion
You can also evaluate xref:manage:monitoring.adoc[metrics] to determine cluster health. If the cluster has any issues, take the broker out of maintenance mode by running the following command before proceeding with other operations, such as decommissioning or retrying the rolling upgrade:
```

File: modules/upgrade/partials/rolling-upgrades/enable-maintenance-mode.adoc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+endif::[]
+ifdef::rolling-restart[]
+You can also evaluate xref:manage:monitoring.adoc[external metrics] to determine cluster health. If the cluster has any issues, take the broker out of maintenance mode by running the following command before proceeding with other operations, such as decommissioning or retrying the rolling restart:
```
Comment (micheleRP) [inline modules/upgrade/partials/rolling-upgrades/enable-maintenance-mode.adoc]:
```suggestion
You can also evaluate xref:manage:monitoring.adoc[metrics] to determine cluster health. If the cluster has any issues, take the broker out of maintenance mode by running the following command before proceeding with other operations, such as decommissioning or retrying the rolling restart:
```

File: modules/upgrade/partials/rolling-upgrades/rolling-restart-intro.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+NODE-ID  NUM-CORES  MEMBERSHIP-STATUS  IS-ALIVE  BROKER-VERSION
+0        1          active             true      v22.3.11
+1        1          active             true      v22.3.11
+2        1          active             true      v22.3.11
+```
+====
+
+New features in a version are enabled after all brokers in the cluster are upgraded. If problems occur, the upgrade is not committed.
+
+NOTE: Redpanda started supporting xref:develop:consume-data/consumer-offsets.adoc[consumer offsets] in version 22.1. When upgrading from version 21.11 to 22.1, after all brokers are upgraded, Redpanda starts to migrate consumer group topics to `__consumer_offsets`. This takes some time, depending on the data size. Until it finishes, all consumer group-related operations (consume, offset commit, coordinator election) are blocked. The migration to consumer offsets is complete when you see `consumer offset feature enabled` in all brokers.
```
Comment (micheleRP) [inline modules/upgrade/partials/rolling-upgrades/rolling-restart-intro.adoc]:
```suggestion
```
Suggest deleting this entire note. I don't think we'd support anyone upgrading from 21.11 to 22.1 now.

File: modules/manage/pages/cluster-maintenance/rolling-restart.adoc [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+include::upgrade:partial$rolling-upgrades/disable-maintenance-mode.adoc[]
+
+include::upgrade:partial$rolling-upgrades/post-upgrade-tasks.adoc[]
+
+include::upgrade:partial$rolling-upgrades/restart-impact.adoc[leveloffset=+1]
+
+include::shared:partial$suggested-reading.adoc[]
+
+* xref:manage:monitoring.adoc[Set up a real-time dashboard to monitor your cluster health]
```
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/rolling-restart.adoc]:
```suggestion
* xref:manage:monitoring.adoc[]
```

File: modules/ROOT/nav.adoc [hunk 7]
```diff
@@ -181,6 +181,7 @@
 *** xref:manage:cluster-maintenance/manage-throughput.adoc[Manage Throughput]
 *** xref:manage:cluster-maintenance/configure-availability.adoc[Configure Availability]
 *** xref:manage:cluster-maintenance/cluster-diagnostics.adoc[Cluster Diagnostics]
+*** xref:manage:cluster-maintenance/rolling-restart.adoc[Perform Rolling Restart]
```
Comment (micheleRP) [inline modules/ROOT/nav.adoc]: @kbatuigas It looks a little odd to have Perform a Rolling Restart in the nav tree, title = Rolling Restart with a subheading Perform a Rolling Restart.

**Comment by kbatuigas** [comment]
@micheleRP Added [new issue](https://github.com/redpanda-data/documentation-private/issues/2136) for creating Rolling restart doc for Kubernetes

**Review by micheleRP — APPROVED** [review]
great update!

**Pull request merged**